hera_qm/xrfi.py

```diff
@@ -607,11 +685,215 @@ def xrfi_h1c_pipe(uv, Kt=8, Kf=8, sig_init=6., sig_adj=2., px_threshold=0.2,
+            Flags of visibilities formed with these antennas will be set to True.
+        metrics_file (str, optional): Metrics file that contains a list of excluded
+            antennas. Flags of visibilities formed with these antennas will be set to True.
+        cal_ext (str, optional): Extension to replace penultimate extension in
+            calfits file for output calibration including flags. Defaults is "flagged_abs".
+            For example, an input_cal of "foo.goo.calfits" would result in
+            "foo.flagged_abs.calfits".
+    Returns:
+        None
+    """
+    history = 'Flagging command: "' + history + '", Using ' + hera_qm_version_str
+    dirname = resolve_xrfi_path(xrfi_path, data_file)
+    xants = process_ex_ants(ex_ants=ex_ants, metrics_file=metrics_file)
+
+    # Initial run on cal data products
+    alg = 'detrend_medfilt'
+    # Calculate metric on abscal data
+    uvc_a = UVCal()
+    uvc_a.read_calfits(acalfits_file)
+    uvf_apriori = UVFlag(uvc_a, mode='flag', copy_flags=True)
+    uvf_ag, uvf_agf = xrfi_pipe(uvc_a, alg=alg, Kt=kt_size, Kf=kf_size, xants=xants,
+                                cal_mode='gain', sig_init=sig_init, sig_adj=sig_adj)
+    uvf_ax, uvf_axf = xrfi_pipe(uvc_a, alg=alg, Kt=kt_size, Kf=kf_size, xants=xants,
+                                cal_mode='tot_chisq', sig_init=sig_init, sig_adj=sig_adj)
+
+    # Calculate metric on omnical data
+    uvc_o = UVCal()
+    uvc_o.read_calfits(ocalfits_file)
+    flag_apply(uvf_apriori, uvc_o, keep_existing=True)
+    uvf_og, uvf_ogf = xrfi_pipe(uvc_o, alg=alg, Kt=kt_size, Kf=kf_size, xants=xants,
+                                cal_mode='gain', sig_init=sig_init, sig_adj=sig_adj)
+    uvf_ox, uvf_oxf = xrfi_pipe(uvc_o, alg=alg, Kt=kt_size, Kf=kf_size, xants=xants,
+                                cal_mode='tot_chisq', sig_init=sig_init, sig_adj=sig_adj)
+
+    # Calculate metric on model vis
+    uv_v = UVData()
+    uv_v.read(model_file)
+    uvf_v, uvf_vf = xrfi_pipe(uv_v, alg=alg, xants=[], Kt=kt_size, Kf=kf_size,
+                              sig_init=sig_init, sig_adj=sig_adj)
+
+    # Combine the metrics together
+    uvf_metrics = uvf_v.combine_metrics([uvf_og, uvf_ox, uvf_ag, uvf_ax],
+                                        method='quadmean', inplace=False)
+    alg_func = algorithm_dict[alg]
+    uvf_metrics.metric_array[:, :, 0] = alg_func(uvf_metrics.metric_array[:, :, 0],
+                                                 flags=~uvf_metrics.weights_array[:, :, 0].astype(np.bool),
+                                                 Kt=kt_size, Kf=kf_size)
+
+    # Flag on combined metrics
+    uvf_f = flag(uvf_metrics, nsig_p=sig_init)
+    uvf_fws = watershed_flag(uvf_metrics, uvf_f, nsig_p=sig_adj, inplace=False)
+    # OR everything together for initial flags
+    uvf_apriori.to_waterfall(method='and', keep_pol=False)
+    uvf_init = uvf_fws | uvf_ogf | uvf_oxf | uvf_agf | uvf_axf | uvf_vf | uvf_apriori
```

Comment:
This combination of flags becomes the initial mask for the second round, which includes the raw visibility data.